### PR TITLE
New `OrbitalRotation` decomposition

### PR DESCRIPTION
**Pull request by isaacdevlugt**
**Context:** New decomposition for `OrbitalDecomposition`.

**Description of the Change:** Decomposes into `SingleExcitation` gates instead of more elementary gates (e.g. control gates, `Hadamard`, etc.)

**Benefits:** 

There are two really big advantages to decomposing down into SingleExcitation gates:
- `default.qubit` and `lightning` natively support `SingleExcitation`, so the execution will be faster
- `SingleExcitation` has a parameter-shift rule defined, so less circuits will be executed to get the gradient

**Related GitHub Issues:** https://github.com/PennyLaneAI/pennylane/issues/3169

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -49,6 +49,9 @@ keyword argument when using `GellMann`, which determines which of the 8 Gell-Man
 
 <h3>Improvements</h3>
 
+* `OrbitalRotation` is now decomposed into two `SingleExcitation` operators for faster execution and a more efficient parameter-shift gradient calculation.
```
Comment (obliviateandsurrender) [inline doc/releases/changelog-dev.md:52]:
```suggestion
* `OrbitalRotation` is now decomposed into two `SingleExcitation` operations for faster execution and more efficient parameter-shift gradient calculations on devices that natively support `SingleExcitation`.
```

File: tests/ops/qubit/test_qchem_ops.py [hunk 2]
```diff
@@ -767,50 +767,14 @@ def test_orbital_rotation_generator(self, phi):
...
-            decomposed_matrix = np.linalg.multi_dot(mats)
-            exp = OrbitalRotation(phi)
-
-            assert np.allclose(decomposed_matrix, exp)
+        op = qml.OrbitalRotation(phi, wires=[0, 1, 2, 3])
+        decomposed_matrix = qml.matrix(
+            qml.SingleExcitation(phi, [0, 2]) @ qml.SingleExcitation(phi, [1, 3]),
+            wire_order=[0, 1, 2, 3],
+        )
+        assert np.equal(decomposed_matrix.all(), op.matrix().all())
```
Comment (obliviateandsurrender) [inline tests/ops/qubit/test_qchem_ops.py:777]: Might prefer using `np.array_equal`, `np.allclose` or just one `.all()` at the end.

**Review by obliviateandsurrender — CHANGES_REQUESTED** [review]
Thanks, @isaacdevlugt! Looking good. Just a couple of minor suggestions, and it will be good to go.

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (isaacdevlugt) [reply doc/releases/changelog-dev.md:52]: Nice!

File: tests/ops/qubit/test_qchem_ops.py (hunk 2 above)
Comment (isaacdevlugt) [reply tests/ops/qubit/test_qchem_ops.py:777]: Agreed!

**Review by obliviateandsurrender — APPROVED** [review]

**Comment by cvjjm** [comment]
Thanks for implementing this!

**Pull request merged**